pkg/virt-manifest/defaults.go

```diff
@@ -0,0 +1,56 @@
+package virt_manifest
+
+import (
+	"fmt"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+)
+
+// AddMinimalDomainSpec takes an existing domain spec and ensures
+// That libvirt's minimum required values are assigned a default
+func AddMinimalDomainSpec(domain *v1.DomainSpec) error {
+	if domain.Name == "" {
+		return fmt.Errorf("domain name cannot be empty")
+	}
+
+	if domain.Type == "" {
+		domain.Type = "qemu"
+	}
+
+	if domain.OS.Type.OS == "" {
+		domain.OS.Type.OS = "hvm"
+	}
+
+	if domain.Memory.Unit == "" {
+		domain.Memory.Unit = "KiB"
+	}
+
+	if domain.Memory.Value == 0 {
+		domain.Memory.Value = 8192
+	}
+
+	// TODO: Forcing the emulator might not be right, but we don't support any others
+	domain.Devices.Emulator = "/usr/local/bin/qemu-x86_64"
+
+	if len(domain.Devices.Interfaces) == 0 {
+		domain.Devices.Interfaces = []v1.Interface{
+			{Type: "network", Source: v1.InterfaceSource{Network: "default"}},
+		}
+	}
+
+	if len(domain.Devices.Graphics) == 0 {
+		domain.Devices.Graphics = []v1.Graphics{
+			v1.Graphics{Type: "spice", Listen: v1.Listen{Type: "address", Address: "0.0.0.0"}},
```

Comment:
Why are you adding a device which was not requested? I don't think we do that right now ...